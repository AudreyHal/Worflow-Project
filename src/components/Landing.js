import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getData } from '../redux/actions/action';
import home from '../images/Home.png';
import audit from '../images/User.png';
import settings from '../images/settings.png';
import logout from '../images/logout.png';
import profile from '../images/profile.png';
import Workflow from './Workflow'

class Landing extends Component{
	constructor(props){
		super(props);
		this.props.getData();
		this.state={	
			showComponent : false,
			showLoader: false
	}
}

	handleClick=()=>{
		this.props.getData();	
		if (this.props.data){
			this.setState({showComponent: true })		
		}
		if (!this.props.data){
			this.setState({showLoader: true })		
		}
	}	

	render(){	
		
		return (		
			<div className="landing-page">
				<div className="row full-page-height">
					<div className="col-2 sidebar">
						<div className="profile-container">
							<img src={profile} alt="profile" className="profile"></img>
							<div className="fullname">Jaohn Samue</div>
							<div className="position">Front Desk Officer</div>
							<div className="position">Finance</div>
						</div> 
						<div className="menu-items">
							<div className="menu-item"><img src={home} alt="home" className="menu-icon"></img><span> Dashboard</span></div>
							<div className="menu-item"><img src={audit} alt="audit" className="menu-icon"></img><span> Audit Log</span></div>
							<div className="menu-item" id="settings"><img src={settings} alt="settings" className="menu-icon"></img><span> Settings</span></div>
						</div>
						<div>
						<div className="menu-item" id="logout-container"><img src={logout} alt="logout" className="menu-icon"></img><span>Log Out</span></div>
						</div>
					</div>
					
					<div className="col main-content-container">
						<div className="navbar row">
							
								<div className="col-auto nav-item">Company</div>
								<div className="col-auto nav-item">Users</div>
								<div className="col-auto nav-item">User Roles</div>
								<div className="col-auto nav-item">Department	</div>
								<div className="col-auto nav-item">Units</div>
								<div className="col-auto nav-item">Document className</div>
								<div className="col-auto nav-item" id="workflow-link" onClick={this.handleClick}><a >Workflows</a></div>
							
						</div>						
						
						{this.state.showComponent ? (	<Workflow ></Workflow> ) : null} 
						{this.state.showLoader ? 
						(
							<div className="text-center full-width row d-flex justify-content-center">          		
								<div className="error-text">Make sure you are connected to the internet and try again.</div>
							</div>
						) : null} 			
						
					</div>    
				</div>
			</div>
		)
	}
}


const mapStateToProps = state => {
	return {
	  data: state.workflows
	};
};  

export default connect(mapStateToProps, {getData: getData})(Landing);