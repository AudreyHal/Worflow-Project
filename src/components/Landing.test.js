import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from "redux-mock-store";
import Landing from "./Landing";
import "../../setupEnzymeTest";
it('should render correctly <Landing>', () => {
  
  const mockStore = configureStore();
  const store = mockStore({
           workflows: [{
               name: "Approval Rating",
               stages: 3
            }],
          loading: false
        });
  const wrapper = mount(
          <Provider store={store}>
              <Landing/>
          </Provider>);
  expect(wrapper).toMatchSnapshot();
});