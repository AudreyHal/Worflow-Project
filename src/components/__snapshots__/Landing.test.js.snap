// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <Landing> 1`] = `ReactWrapper {}`;
